logging/logger.go

```diff
@@ -111,9 +111,8 @@ func (logger *zLogger) AddAppender(appender Appender) {
 	// Not supported
 }
 
-func (logger zLogger) WithFields(args ...interface{}) Logger {
-	// Not supported. Use With() instead.
-	return nil
+func (logger *zLogger) WithFields(args ...interface{}) Logger {
+	return &zLogger{logger.AsZap().With(args...)}
```

Comment:
i'm going to move this discussion to [this thread](https://github.com/viamrobotics/rdk/pull/4475#discussion_r1813091766) because the context behind this change is there.

Reply:
@maximpertsov showed me what he was observing. The `TestModuleDebugReconfigure` test is using [this helper](https://github.com/viamrobotics/rdk/blob/8e378dbf990490b584ddfab4ef3f12f67322070e/testutils/log_utils.go?plain=1#L22) which definitely uses this code.

No objection for keeping this change in.